components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.rdbms/src/main/java/org/wso2/carbon/event/output/adapter/rdbms/RDBMSEventAdapter.java

```diff
@@ -0,0 +1,591 @@
+    }
+
+    @Override
+    public void testConnect() throws TestConnectionNotSupportedException {
+
+        try {
+            DataSource dataSource;
+            CarbonDataSource carbonDataSource = RDBMSEventAdapterServiceValueHolder.getDataSourceService().
+                    getDataSource(eventAdapterConfiguration.getStaticProperties().
+                            get(RDBMSEventAdapterConstants.ADAPTER_GENERIC_RDBMS_DATASOURCE_NAME));
+            if (carbonDataSource != null) {
+                dataSource = (DataSource) carbonDataSource.getDSObject();
+                con = dataSource.getConnection();
+            } else {
+                throw new OutputEventAdapterRuntimeException("There is no any datasource found named "
+                        + RDBMSEventAdapterConstants.ADAPTER_GENERIC_RDBMS_DATASOURCE_NAME + " to connect.");
+            }
+        } catch (Exception e) {
+            throw new OutputEventAdapterRuntimeException(e);
+        } finally {
+            cleanupConnections(null, con);
+        }
+    }
+
+    @Override
+    public void connect() {
+
+        try {
+            CarbonDataSource carbonDataSource = RDBMSEventAdapterServiceValueHolder.getDataSourceService().getDataSource(
+                    eventAdapterConfiguration.getStaticProperties().get(RDBMSEventAdapterConstants
+                            .ADAPTER_GENERIC_RDBMS_DATASOURCE_NAME));
+            con = ((DataSource) carbonDataSource.getDSObject()).getConnection();
+        } catch (DataSourceException e) {
+            log.error(
+                    "There is no any data-source found called : " + eventAdapterConfiguration.getStaticProperties().get(
+                            RDBMSEventAdapterConstants
+                                    .ADAPTER_GENERIC_RDBMS_DATASOURCE_NAME), e);
+            throw new ConnectionUnavailableException(e.getMessage(), e);
+        } catch (SQLException e) {
+            throw new ConnectionUnavailableException(e);
+        } finally {
+            cleanupConnections(null, con);
+        }
+
+    }
+
+    @Override
+    public void publish(Object message, Map<String, String> dynamicProperties) {
+
+        String tableName;
+
+        try {
+            if (message instanceof Map) {
+
+                tableName = eventAdapterConfiguration.getStaticProperties().get(RDBMSEventAdapterConstants.ADAPTER_GENERIC_RDBMS_TABLE_NAME);
+                String executionMode = eventAdapterConfiguration.getStaticProperties().get(RDBMSEventAdapterConstants
+                        .ADAPTER_GENERIC_RDBMS_EXECUTION_MODE);
+                String updateColumnKeys = eventAdapterConfiguration.getStaticProperties().get(RDBMSEventAdapterConstants
+                        .ADAPTER_GENERIC_RDBMS_UPDATE_KEYS);
+
+                if (executionInfo == null) {
+                    executionInfo = new ExecutionInfo();
+                    initializeDatabaseExecutionInfo(tableName, executionMode, updateColumnKeys, message, executionInfo);
```

Comment:
Please fix the bug: executionInfo = new ExecutionInfo(); should not be initialised here